dask/array/tests/test_creation.py

```diff
@@ -591,13 +597,29 @@ def test_tile_neg_reps(shape, chunks, reps):
 
 
 @pytest.mark.parametrize("shape, chunks", [((10,), (1,)), ((10, 11, 13), (4, 5, 3))])
-@pytest.mark.parametrize("reps", [[1], [1, 2]])
-def test_tile_array_reps(shape, chunks, reps):
+@pytest.mark.parametrize("reps", [0, (0,), (2, 0), (0, 3, 0, 4)])
+def test_tile_zero_reps(shape, chunks, reps):
     x = np.random.random(shape)
     d = da.from_array(x, chunks=chunks)
 
-    with pytest.raises(NotImplementedError):
-        da.tile(d, reps)
+    assert_eq(np.tile(x, reps), da.tile(d, reps))
+
+
+@pytest.mark.parametrize("a", [da.asarray([[[]]]), da.asarray([[], []])])
+@pytest.mark.parametrize("reps", [2, (3, 2, 5)])
+def test_tile_empty_array(a, reps):
+    assert_eq(np.tile(a, reps), da.tile(a, reps))
```

Comment:
```suggestion
    assert_eq(np.tile(a.compute(), reps), da.tile(a, reps))
```

Reply:
As of Numpy 1.17 `np.tilie` will just dispatch to `da.tile` if given a dask array as an input, so let's coerce to numpy here first.

Reply:
That's a cool feature I didn't know about! Changed now.